spec/haskell/src/SEL4/Object/TCB/AARCH64.hs

```diff
@@ -51,3 +53,12 @@ getSanitiseRegisterInfo t = do
 
 postModifyRegisters :: PPtr TCB -> PPtr TCB -> UserMonad ()
 postModifyRegisters _ _ = return ()
+
+postSetFlags :: PPtr TCB -> TcbFlags -> Kernel ()
+postSetFlags t flags =
+    when (tcbFlagToWord (ArchFlag FpuDisabled) .&. flags /= 0) (fpuRelease t)
```

Comment:
interesting, with one flag I guess this is the shortest way to do it, but with more flags it'll be a bunch of `tcbFlagToWord`s `.|.`'ed together?

Reply:
Hmm, good question, I hadn't thought about what this will look like if/when there are more flags. Maybe it's worth adding some sort of `flagSet` helper function that does the `tcbFlagToWord ` and `.&.`. That would make this something like 
```
when (flagSet (ArchFlag FpuDisabled) flags) (fpuRelease t)
```

Reply:
Something like that sounds good, but I'd prefer `isFlagSet` or something, otherwise it sounds like a flag-setter.

Reply:
Maybe `inFlagSet`?

Reply:
I like `isFlagSet`, to me it's more of an abstraction where we don't care how it's implemented while `inFlagSet` implies that it's a set.

Reply:
`hasFlag`? I keep reading `isFlagSet` as "is this a flag set", not "is this flag set/unset".

Reply:
Uhh, I really think that `isFlagSet` is pretty standard for "is the flag set", I'm not sure many people would think of `isFlagASet` when reading it, and I'm not even sure that's possible in Isabelle's type system, and `isSetOfFlags` is an even more creative interpretation...

Reply:
I still prefer `isFlagSet` out of these options. @lsf37 have a look at https://github.com/seL4/l4v/pull/819/commits/52a23eb5c4298aa8a6a6295975cd62c8b64050fc and see whether seeing it in context looks better to you or not.